spikeinterface/preprocessing/silence_periods.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+import scipy.interpolate
+
+from spikeinterface.core.core_tools import define_function_from_class
+from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+
+from ..core import get_random_data_chunks, get_noise_levels
+
+class SilencedPeriodsRecording(BasePreprocessor):
+    """
+    Silence user-defined periods from recording extractor traces. By default, 
+    periods are zeroed-out (mode = 'zeros'). This is only recommended 
+    for traces that are centered around zero (e.g. through a prior highpass
+    filter); if this is not the case, you can also fill the periods with noise.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor to remove artifacts from
+    list_periods: list of lists/arrays
+        One list per segment of tuples (start_frame, end_frame) to silence
+
```

Comment:
I think that period should be a list of numpy arrays always for efficiency.

Reply:
I reuse the same syntax as in remove_artifacts. It can be arrays or list. You want to have a list of arrays?